libs/ngxtension/inject-query-params/src/inject-query-params.ts

```diff
@@ -1,31 +1,44 @@
-import { assertInInjectionContext, inject, type Signal } from '@angular/core';
+import { inject, type Signal } from '@angular/core';
 import { toSignal } from '@angular/core/rxjs-interop';
 import { ActivatedRoute, type Params } from '@angular/router';
+import { assertInjector } from 'ngxtension/assert-injector';
+import {
+	DefaultValueOptions,
+	InjectorOptions,
+	ParseOptions,
+} from 'ngxtension/shared';
 import { map } from 'rxjs';
 
 type QueryParamsTransformFn<ReadT> = (params: Params) => ReadT;
 
 /**
- * The `InputOptions` interface defines options for configuring the behavior of the `injectQueryParams` function.
+ * The `QueryParamsOptions` type defines options for configuring the behavior of the `injectQueryParams` function.
  *
  * @template ReadT - The expected type of the read value.
  * @template WriteT - The type of the value to be written.
- * @template InitialValueT - The type of the initial value.
+ * @template DefaultValueT - The type of the default value.
  */
-export interface QueryParamsOptions<ReadT, WriteT, InitialValueT> {
-	/**
-	 * A transformation function to convert the written value to the expected read value.
-	 *
-	 * @param v - The value to transform.
-	 * @returns The transformed value.
-	 */
-	transform?: (v: WriteT) => ReadT;
-
-	/**
-	 * The initial value to use if the query parameter is not present or undefined.
-	 */
-	initialValue?: InitialValueT;
-}
+export type QueryParamsOptions<ReadT, DefaultValueT> = ParseOptions<
+	ReadT,
+	string | null
+> &
+	DefaultValueOptions<DefaultValueT> &
+	InjectorOptions & {
+		/**
+		 * The initial value to use if the query parameter is not present or undefined.
+		 *
+		 * @deprecated Use `defaultValue` as a replacement.
+		 */
+		initialValue?: DefaultValueT;
```

Comment:
We can write a migration schematic when we remove this one. So it's smooth for everyone when they update.

Reply:
So you mean we shouldn't mark it as deprecated?

Reply:
No marking them as deprecated is fine.